distributed/worker.py

```diff
@@ -735,10 +735,13 @@ def local_dir(self):
         )
         custom = {}
         for k, metric in self.metrics.items():
-            result = metric(self)
-            if hasattr(result, "__await__"):
-                result = await result
-            custom[k] = result
+            try:
+                result = metric(self)
+                if hasattr(result, "__await__"):
+                    result = await result
+                custom[k] = result
+            except Exception:
```

Comment:
Can we catch only the exception(s) raised in the situation where it's safe to ignore them? This catch-all handler will catch syntax errors and broken things that should work, making for a potential debugging mystery.

Reply:
Good point.   Unfortunately the `metrics` provided here can be user provided, so we don't have a strong sense of what exactly we can catch and not catch.

Normally I would handle this by logging the error properly, so that we at least report up what's wrong while we don't error.  However, this gets called frequently, and I would like to avoid dumping a ton of logging messages.  This could be resolved by having some sort of log-once check or state, but at the moment I'm a bit pressed and didn't want to build this.

Suggestions welcome.  Otherwise I'm going to merge and release shortly (there are a couple of other things that I'd like to get out there as well).

Reply:
Okay, that's reasonable IMO.  In lieu of your logging solution, maybe a FIXME describing that would be appropriate, possibly even with a link to your response here.